pkg/providers/amifamily/resolver.go

```diff
@@ -54,10 +54,10 @@ type Options struct {
 	InstanceProfile         string
 	CABundle                *string `hash:"ignore"`
 	// Level-triggered fields that may change out of sync.
-	SecurityGroupsIDs []string
-	Tags              map[string]string
-	Labels            map[string]string `hash:"ignore"`
-	KubeDNSIP         net.IP
+	SecurityGroups []*ec2.SecurityGroup
```

Comment:
All the details in the options are stored in terms of their internal API concept, rather than basing it on the `ec2` version of the data. i.e. Can we store the data in the options as `v1alpha1.SecurityGroup` and then transform the data to the ec2 struct when we are creating the launch template